### DISCUSSION TITLE
User-locked license stored in plain text?

### DISCUSSION THREAD

**Discussion by dizcza** [keygen-sh/community#59] [Q&A] [answered]
I'm writing a script to validate either online or offline license.
I have a question regarding the User-locked license & POST request. It seems that the passed "key", the license itself, is not required to be additionally cryptographically encrypted on the user's machine, is it? And can be stored in a file as plain text. Is it more like a public SSH key then?
If storing the license (f.i., `C1B6DE-39A6E3-DE1529-8559A0-4AF593-V3` from your examples) on the user's storage requires additional encryption, what's the recommended procedure?
Thanks.

**Comment by ezekg** [comment, marked as answer]
This will depend on your application's security model. Typically, storing or caching the license key in plain text is fine, since it only acts as a local decryption key to the license file which is itself encrypted.
